docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,324 @@
+    imagenet_std = [0.229, 0.224, 0.225]  # std of the ImageNet dataset for normalizing
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print('The running processor is...', device) 
+
+CelebA Dataset
+==============
+
+We are going to use CelebA dataset for this experiment. The CelebA dataset  is a large-scale face attributes dataset which can be employed as the training and test sets for the following computer vision tasks: face attribute recognition, face detection, landmark (or facial part) localization, and face editing & synthesis.
+
+Fetching data 
+=============
+
+The section below consists of a few lines of codes that help us download celebA dataset from a public web source and unzip them.
+
+.. code-block:: python
+
+    data_root = 'data/celeba'
+    # Path to folder with the dataset
+    dataset_folder = f'{data_root}/img_align_celeba'
+
+    # URL for the CelebA dataset (aligned images, attributes, landmasrks)
+    url = 'https://graal.ift.ulaval.ca/public/celeba/img_align_celeba.zip'
+    attr_url = 'https://graal.ift.ulaval.ca/public/celeba/list_attr_celeba.txt'
+    land_mark_url = 'https://graal.ift.ulaval.ca/public/celeba/list_landmarks_align_celeba.txt'
+
+    # Path to download the dataset to
+    download_path = f'{data_root}/img_align_celeba.zip'
+    land_mark_path = f'{data_root}/list_landmarks_align_celeba.txt'
+    attr_path = f'{data_root}/list_attr_celeba.txt'
+
+    # Create required directories 
+    if not os.path.exists(data_root):
+    os.makedirs(data_root)
+    os.makedirs(dataset_folder)
+
+    # Download the dataset from the source
+    gdown.download(url, download_path, quiet=False)
+    gdown.download(land_mark_url, land_mark_path, quiet=False)
+    gdown.download(attr_url, attr_path, quiet=False)
+
+    # Unzip the downloaded file 
+    with zipfile.ZipFile(download_path, 'r') as ziphandler:
+    ziphandler.extractall(dataset_folder)
+   
+Create a custom dataset class
+=============================
+
+As we are going to implement a multi-task problem by a single CNN, we should provide the CNN with the ground truth in a proper way. Here, we have two different tasks: classification and regression. In the classification task, the goal is to identify the gender. The labels of the gender for each image are saved in the `list_attr_celeba.txt` file, in which 1 stands for male and -1 for female. Since we consider the loss of both tasks simultaneously, we scale all target values to the range of [0,1]. Hence, the gender labels will be changed as well, 1 for male and 0 for female. For the localization part, the coordinates of the eyes (Left and Right) are provided in the `list_landmarks_align_celeba.txt` file. In addition to scaling the number to the range of [0,1], we also need to rescale the coordinates to the image's new size (224,224).
+
+.. code-block:: python
+
+    class CelebADataset(Dataset):
```

Comment:
Why don't you use [torchvision's CelabA dataset](https://pytorch.org/vision/stable/datasets.html#celeba) instead?

Reply:
@mohammad-brdrn You can download the dataset like you do before instantiating the torchvision dataset. In that way, it won't try to download it from Internet.

Reply:
you mean using torchvision.datasets.celebA(root=root, download=True)?
If so, that's the problem I mentioned. it always fails to download. also, torch communities also admit that and suggest downloading from the internet. for example link below:

https://github.com/pytorch/vision/issues/2262

Reply:
Documentation says:

> download (bool, optional) – If true, downloads the dataset from the internet and puts it in root directory. If dataset is already downloaded, it is not downloaded again.

So if you download the dataset via the current links and put the files under the file hierarchy expected by torchvision.datasets.CelebA, it will work.

Reply:
I put several days on this trying different strategies but still, we cannot use torch.datasets.CelbA as the standard dataset. because:

1)we publish this example mainly on google collab for the readers. Googlecolab does not keep downoladed files after runtime. so the problem still exists.
2) I tried downloading the file (first by setting download=true) and using it with (download=False). I tried it more than 9 days at almost any time (even midnight) to gives the same error.
3) I read almost all relevant posts on this issue and also inspected the root after the error. in fact, after the error (BadZipFile: File is not a zip file) i saw that it has downloaded all the files!! but it is not able to unzip the file! i tried downloading and unzipping the zipfile with other ways, but it doesnt work again.
4) can we proceed with the proposed dataset class? in meanwhile i will try other ways as well. i don't want to stop other parts for this long time, focusing on this part. maybe i can resume it along with other parts.

<img width="444" alt="3afterzipls" src="https://user-images.githubusercontent.com/50740990/119734646-48e83680-be49-11eb-8734-fd9b3e525c3b.png">
<img width="491" alt="1datadown" src="https://user-images.githubusercontent.com/50740990/119734731-60bfba80-be49-11eb-9ddc-c5635f96e5b4.png">
<img width="391" alt="2zip" src="https://user-images.githubusercontent.com/50740990/119734749-6a492280-be49-11eb-9a73-bb75a4f447ee.png">

Reply:
Hi @mohammad-brdrn , sorry for the late response. Here is some code which works for using the CelebA dataset:

For loading the files:
```python
data_root = 'datasets'

base_url = "https://graal.ift.ulaval.ca/public/celeba/"

file_list = [
    "img_align_celeba.zip",
    "list_attr_celeba.txt", 
    "identity_CelebA.txt",
    "list_bbox_celeba.txt",
    "list_landmarks_align_celeba.txt",
    "list_eval_partition.txt"
]

# Path to folder with the dataset
dataset_folder = f'{data_root}/celeba'
os.makedirs(dataset_folder, exist_ok=True)

for file in file_list:
    url = f"{base_url}/{file}"
    wget.download(url, f"{dataset_folder}/{file}")

with zipfile.ZipFile(f"{dataset_folder}/img_align_celeba.zip", 'r') as ziphandler:
    ziphandler.extractall(dataset_folder)
```
Then, instantiating the datasets:
```python
train_dataset = datasets.CelebA(data_root, 
                                split='train', 
                                target_type=['attr', 'landmarks'], 
                                transform=tfms.ToTensor())
valid_dataset = datasets.CelebA(data_root, 
                                split='valid', 
                                target_type=['attr', 'landmarks'], 
                                transform=tfms.ToTensor())
test_dataset = datasets.CelebA(data_root, 
                               split='test', 
                               target_type=['attr', 'landmarks'], 
                               transform=tfms.ToTensor())
```

You had to look in the [code of CelebA](https://pytorch.org/vision/stable/_modules/torchvision/datasets/celeba.html#CelebA) to know how to do it.

Reply:
Btw, when using wget, you should verify that the files do not exist before downloading. Otherwise, it creates files with a suffix "(1)".

Reply:
I applied the changes ( using torch.datasets.CelebA), While you check the overall format, I may work more on hyperparameters.